File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        int [] ans = new int[2];
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(target-nums[i])){
                int ans2 = map.get(target-nums[i]);
                int ans1=i;
                ans[0] = ans2;
                ans[1] = ans1;
                // Arrays.sort(ans);
                return ans;
            }
            
            map.put(nums[i],i);
        }

        return ans;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-3.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        di = dict()
        for i in range(len(nums)):
            di[nums[i]] = di.get(nums[i],0)+1
        for n in di.values():
            if(n>=2):
                return True
        
        return False
        